### PR TITLE
Port the UFS to S4

### LINKED ISSUE #738: Building and running the model on S4
## Description
To support development and data assimilation efforts, the UFS model should be ported to run on SSEC's S4 system.

## Solution
The machine-related scripts should be updated to include S4 and additional module files should be added to support S4 builds.

## Related to
Work is underway to port the entire global workflow: NOAA-EMC/global-workflow#138

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR. (N/A)

## Description

Adds support for the S4 supercomputer to the UFS.

### Issue(s) addressed

- fixes #738 
- references noaa-emc/global-workflow/issues/138

## Testing

The regression tests have all passed on S4 with the Intel compilers using develop_20210901.  Another set of tests are underway using develop_20210907 and I will commit that log when finished.

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] CI
- [ ] S4

## Dependencies

- Waiting on #790

### CODE REVIEW

File: cmake/Intel.cmake
Comment (climbfuji) [inline cmake/Intel.cmake]: Rather than specifying a specific Intel architecture, would it be better to use the flags for AVX (that is, the original AVX, not AVX2) here?

File: tests/rt.sh
Comment (climbfuji) [inline tests/rt.sh]: Is it possible to configure `ecflow` in addition to `rocoto`?

File: tests/rt.sh
Comment (climbfuji) [inline tests/rt.sh]: Why is this being updated? Do we have the updated data set on the other platforms? What is different to the 20210621 input data?

File: cmake/configure_s4.intel.cmake
Comment (climbfuji) [inline cmake/configure_s4.intel.cmake]: I think that `IVYBRIDGE` (or preferably `AVX`) should be added to the top-level `CMakeLists.txt` as well.

**Review by climbfuji — CHANGES_REQUESTED** [review]
Thanks for getting the regression tests to work (again) on S4. I have a general question. What are we going to do after this PR is merged? Will S4 become a tier-1 platform, i.e. does it need to pass all regression tests before PRs are merged? Or will it be a tier-2 platform (test post-commit, either every time or regularly every week, 2 weeks, month)? Who is taking responsibility for maintaining S4? Maintaining and testing platforms does require some time commitment. Thanks!

**Comment by BrianCurtis-NOAA** [comment]
The `RegressionTests_s4.intel.log` file looks like it is from the baseline creation `./rt.sh -c` . If you haven't yet, make sure to run the regression tests against those newly created baselines (without `-c`). We also typically use ecflow (although not mandatory) and the AutoRT script uses it (if it's to be a Tier-1 platform).

**Comment by DavidHuber-NOAA** [comment]
@climbfuji S4 will be a tier-2 system.  I can commit to testing it monthly.  I am the science integrator on S4, so it's my responsibility to maintain the packages and ports on the system.

@BrianCurtis-NOAA I will make sure to run without the -c, thanks.  Should the -c option be run every time there is an update to inputs?  Regarding `ecflow`, it is installed on S4, though I haven't tested it.  I will give it a try and add support if it works.

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA I will make sure to run without the -c, thanks. Should the -c option be run every time there is an update to inputs? Regarding `ecflow`, it is installed on S4, though I haven't tested it. I will give it a try and add support if it works.

Any time the baselines change from a PR, we create new baselines and run the regression tests on them afterwards.

File: tests/rt.sh
Comment (DavidHuber-NOAA) [reply tests/rt.sh]: This is the dataset that is being used to test #790, which this PR is reliant on.  That data has been added to at least Hera, Gaea, Orion, and WCOSS-Dell.  I will defer to @aliabdolali regarding the difference in the data.

Comment (JessicaMeixner-NOAA) [reply tests/rt.sh]: These are updated mod_def files for waves for the updated code.

Comment (climbfuji) [reply tests/rt.sh]: Ok, thanks. We will need to copy these data over to Jet and Cheyenne.

File: cmake/Intel.cmake
Comment (DavidHuber-NOAA) [reply cmake/Intel.cmake]: OK, I've updated switched to using AVX instead.

File: tests/rt.sh
Comment (DavidHuber-NOAA) [reply tests/rt.sh]: I've added support for `ecflow` to rt.sh and I did a quick test using it.  I'll do a full set of tests with it as well.

File: cmake/configure_s4.intel.cmake
Comment (DavidHuber-NOAA) [reply cmake/configure_s4.intel.cmake]: I've made the change here to `AVX` as well as the top-level `CMakeLists.txt`.

**Review by climbfuji — APPROVED** [review]
Thanks for addressing my comments, this looks good to me. We just need to remember to copy the new WW3 baselines across.

**Comment by climbfuji** [comment]
@DavidHuber-NOAA @JessicaMeixner-NOAA the new WW3 data `WW3_input_data_20210908` is now on Jet and Cheyenne inside the `input-data-20210825` directory.

**Comment by climbfuji** [comment]
> @DavidHuber-NOAA @JessicaMeixner-NOAA the new WW3 data `WW3_input_data_20210908` is now on Jet and Cheyenne inside the `input-data-20210825` directory.

@MinsukJi-NOAA I also added the new directory `WW3_input_data_20210908` to the `input-data-20210825` directory on S3.

**Comment by DusanJovic-NOAA** [comment]
Please update your branches, resolve conflicts. We will combine this PR with #812

**Comment by DavidHuber-NOAA** [comment]
@DusanJovic-NOAA The branch is now up to date.

**Comment by DavidHuber-NOAA** [comment]
Merged via #812.

**Pull request closed without merging**
